FreeRTOS/FreeRTOS
# change(freertos/smp/granular_locks): Updated prvYieldCore() coverage tests

Description
-----------
This commit updates the code coverage tests for `prvYieldCore()` in SMP configuration, viz., `test_coverage_prvYieldCore_core_id_ne_current_coreid` and `test_coverage_prvYieldCore_runstate_eq_yielding` by adding a unit test configuration to force preemption disable for the task invoking the coverage test.

This change is done to fix the CI test failures observed in the FreeRTOS-Kernel [PR#1154](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/1154). The failure is observed in the GitHub action [here](https://github.com/sudeep-mohanty/FreeRTOS-Kernel/actions/runs/11612132105/job/32334942902).
The change is necessary because the granular locks PR updates `xPreemptionDisable` to be nested. Reference [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/1154/commits/e223842bf4292a9e33c68c396f4d5668e5cc95eb). Since the `xPreemptionDisable` is not set in the test, the call to `vTaskPreemptionEnable()` results in undefined behavior.

The change would not affect the current kernel and CI tests as `xPreemptionDisable` is overwritten in the `vTaskPreemptionEnable()` call in the tests that have been updated in this PR.

Test Steps
-----------
- FreeRTOS CI runs for the [PR#1154](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/1154).

Checklist:
----------

Related Issue
-----------
- Relates to https://github.com/FreeRTOS/FreeRTOS-Kernel/issues/905

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
